argon2id.go

```diff
@@ -167,18 +168,35 @@ func DecodeHash(hash string) (params *Params, salt, key []byte, err error) {
 	}
 
 	params = &Params{}
-	_, err = fmt.Sscanf(vals[3], "m=%d,t=%d,p=%d", &params.Memory, &params.Iterations, &params.Parallelism)
+	_, err = fmt.Fscanf(r, "m=%d,t=%d,p=%d$", &params.Memory, &params.Iterations, &params.Parallelism)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	rest, err := ioutil.ReadAll(r)
```

Comment:
`ioutil` is deprecated, could use `io.ReadAll` if we can assume Go 1.16+. The only info in the package I could locate about the supported version is 1.12 in `go.mod`.